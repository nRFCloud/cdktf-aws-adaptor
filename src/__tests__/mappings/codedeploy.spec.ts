import { CodedeployDeploymentGroup } from "@cdktf/provider-aws/lib/codedeploy-deployment-group";
import { CfnDeploymentGroup } from "aws-cdk-lib/aws-codedeploy";
import { synthesizeElementAndTestStability } from "../helpers.js";

describe("CodeDeploy mappings", () => {
  it("should map CfnFunction to LambdaFunction", () => {
    synthesizeElementAndTestStability(
      CfnDeploymentGroup,
      {
        applicationName: "my-application",
        deploymentConfigName: "CodeDeployDefault.LambdaAllAtOnce",
        deploymentGroupName: "my-deployment-group",
        serviceRoleArn: "arn:aws:iam::123456789012:role/CodeDeployServiceRole",
        autoRollbackConfiguration: {
          enabled: true,
          events: ["DEPLOYMENT_FAILURE"],
        },
        deploymentStyle: {
          deploymentOption: "WITH_TRAFFIC_CONTROL",
          deploymentType: "BLUE_GREEN",
        },
        ec2TagFilters: [
          {
            key: "my-key",
            type: "KEY_AND_VALUE",
            value: "my-value",
          },
        ],
        ec2TagSet: {
          ec2TagSetList: [
            {
              ec2TagGroup: [
                {
                  key: "my-key",
                  type: "KEY_AND_VALUE",
                  value: "my-value",
                },
              ],
            },
          ],
        },
        ecsServices: [
          {
            clusterName: "my-cluster",
            serviceName: "my-service",
          },
        ],
        loadBalancerInfo: {
          elbInfoList: [
            {
              name: "my-elb",
            },
          ],
        },
        onPremisesInstanceTagFilters: [
          {
            key: "my-key",
            type: "KEY_AND_VALUE",
            value: "my-value",
          },
        ],
        triggerConfigurations: [
          {
            triggerEvents: ["DeploymentFailure"],
            triggerName: "my-trigger",
            triggerTargetArn: "arn:aws:sns:us-east-1:123456789012:my-topic",
          },
        ],
      },
      CodedeployDeploymentGroup,
      {
        appName: "my-application",
        deploymentConfigName: "CodeDeployDefault.LambdaAllAtOnce",
        deploymentGroupName: "my-deployment-group",
        serviceRoleArn: "arn:aws:iam::123456789012:role/CodeDeployServiceRole",
        autoRollbackConfiguration: {
          enabled: true,
          events: ["DEPLOYMENT_FAILURE"],
        },
        deploymentStyle: {
          deploymentOption: "WITH_TRAFFIC_CONTROL",
          deploymentType: "BLUE_GREEN",
        },
        ec2TagFilter: [
          {
            key: "my-key",
            type: "KEY_AND_VALUE",
            value: "my-value",
          },
        ],
        ec2TagSet: [{
          ec2TagFilter: [
            {
              key: "my-key",
              type: "KEY_AND_VALUE",
              value: "my-value",
            },
          ],
        }],
        ecsService: {
          clusterName: "my-cluster",
          serviceName: "my-service",
        },
        loadBalancerInfo: {
          elbInfo: [
            {
              name: "my-elb",
            },
          ],
        },
        onPremisesInstanceTagFilter: [
          {
            key: "my-key",
            type: "KEY_AND_VALUE",
            value: "my-value",
          },
        ],
        triggerConfiguration: [
          {
            triggerEvents: ["DeploymentFailure"],
            triggerName: "my-trigger",
            triggerTargetArn: "arn:aws:sns:us-east-1:123456789012:my-topic",
          },
        ],
      },
    );
  });
});
