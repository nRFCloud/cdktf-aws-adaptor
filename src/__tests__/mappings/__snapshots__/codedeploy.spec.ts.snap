// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CodeDeploy mappings > should map CfnFunction to LambdaFunction 1`] = `
"{
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"alias\\": \\"us_east_1\\",
        \\"region\\": \\"us-east-1\\"
      }
    ]
  },
  \\"resource\\": {
    \\"aws_codedeploy_deployment_group\\": {
      \\"resource_22C949BF\\": {
        \\"app_name\\": \\"my-application\\",
        \\"auto_rollback_configuration\\": {
          \\"enabled\\": true,
          \\"events\\": [
            \\"DEPLOYMENT_FAILURE\\"
          ]
        },
        \\"deployment_config_name\\": \\"CodeDeployDefault.LambdaAllAtOnce\\",
        \\"deployment_group_name\\": \\"my-deployment-group\\",
        \\"deployment_style\\": {
          \\"deployment_option\\": \\"WITH_TRAFFIC_CONTROL\\",
          \\"deployment_type\\": \\"BLUE_GREEN\\"
        },
        \\"ec2_tag_filter\\": [
          {
            \\"key\\": \\"my-key\\",
            \\"type\\": \\"KEY_AND_VALUE\\",
            \\"value\\": \\"my-value\\"
          }
        ],
        \\"ec2_tag_set\\": [
          {
            \\"ec2_tag_filter\\": [
              {
                \\"key\\": \\"my-key\\",
                \\"type\\": \\"KEY_AND_VALUE\\",
                \\"value\\": \\"my-value\\"
              }
            ]
          }
        ],
        \\"ecs_service\\": {
          \\"cluster_name\\": \\"my-cluster\\",
          \\"service_name\\": \\"my-service\\"
        },
        \\"load_balancer_info\\": {
          \\"elb_info\\": [
            {
              \\"name\\": \\"my-elb\\"
            }
          ]
        },
        \\"on_premises_instance_tag_filter\\": [
          {
            \\"key\\": \\"my-key\\",
            \\"type\\": \\"KEY_AND_VALUE\\",
            \\"value\\": \\"my-value\\"
          }
        ],
        \\"service_role_arn\\": \\"arn:aws:iam::123456789012:role/CodeDeployServiceRole\\",
        \\"tags\\": {
        },
        \\"trigger_configuration\\": [
          {
            \\"trigger_events\\": [
              \\"DeploymentFailure\\"
            ],
            \\"trigger_name\\": \\"my-trigger\\",
            \\"trigger_target_arn\\": \\"arn:aws:sns:us-east-1:123456789012:my-topic\\"
          }
        ]
      }
    }
  },
  \\"terraform\\": {
    \\"backend\\": {
      \\"local\\": {
        \\"path\\": \\"/terraform.test-stack.tfstate\\"
      }
    },
    \\"required_providers\\": {
      \\"aws\\": {
        \\"source\\": \\"aws\\",
        \\"version\\": \\"5.23.1\\"
      }
    }
  },
  \\"variable\\": {
    \\"resource_refs_CDA17697\\": {
      \\"default\\": \\"\${aws_codedeploy_deployment_group.resource_22C949BF.id}\\",
      \\"type\\": \\"string\\"
    }
  }
}"
`;
